proto/viam/app/data/v1/data.proto

```diff
@@ -47,6 +47,22 @@ enum Order {
   ORDER_ASCENDING = 2;
 }
 
+enum Status {
+  STATUS_UNSPECIFIED = 0;
+  STATUS_PARTIAL_SUCCESS = 1;
+  STATUS_SUCCESS = 2;
+}
+
+// Error message is an aggregated error message string
+message Error {
+  string message = 1;
+}
+
+message Result {
+  Status status = 1;
+  Error error = 2;
```

Comment:
[nit] I think you can just embed the message + comment here, like @agreenb said, since we probably don't need a separate message for error. If we were to want result to be more descriptive, with int code (for status codes), we could just add them directly here.

```
message Result {
  Status status = 1;
  // message is an aggregated error message string
  string message = 2;
}

```

Reply:
alrighty!